### ISSUE TITLE
BonminNLSolver failing on a simple problem

### ISSUE THREAD

**Issue by rb004f** [JuliaOpt/AmplNLWriter.jl#50]
The following (simple) problem fails on windows with "WARNING: Not solved to optimality, status: Error"

```
m2 = Model(solver =  BonminNLSolver(["bonmin.nlp_log_level=1"; "bonmin.bb_log_level=1"]))
@variable(m2, 0 <= yp <= 1, Int)
@variable(m2, 0 <= l <= 1000.0)
@variable(m2, 0 <= f <= 1000.0)
@NLconstraint(m2, .087*l >= f^2)
@constraint(m2,  l <= yp*1000.0)
v = solve(m2)
println(v)
```

However, it succeeds with OSIL

```
m4 = Model(solver =  OsilBonminSolver())
@variable(m4, 0 <= yp <= 1, Int)
@variable(m4, 0 <= l <= 1000.0)
@variable(m4, 0 <= f <= 1000.0)
@NLconstraint(m4, .087*l >= f^2)
@constraint(m4,  l <= yp*1000.0)
v = solve(m4)
println(v)
```

Packages were updated as of Sept. 20, 2016, using the installations of Bonmin built by CoinOptServices and AMPLNLWriter

**Comment by mlubin** [comment]
@rb004f, looks like some of your code got garbled, you should use the raw quote blocks (with 3 backticks) for code.

**Comment by rb004f** [comment]
Yup, the perils of cut and paste.  Let me try again

**Comment by JackDunnNZ** [comment]
@rb004f Thanks for reporting this. There was an issue with some of the statistics in the .nl file header, and this was causing problems for models with no objective function. If you grab the latest code with `Pkg.checkout("AmplNLWriter")` your example will work now.

**Comment by rb004f** [comment]
Thanks @JackDunnNZ!  I think I am still seeing an issue with problems with no objective function.  With that same example above, I see this error message

Got an exception of type MethodError outside of a @test
  MethodError: `+` has no method matching +(::Void, ::Float64)
  Closest candidates are:
    +(::Any, ::Any, !Matched::Any, !Matched::Any...)
    +(!Matched::Float64, ::Float64)
    +(!Matched::Complex{Bool}, ::Real)
    ...
   in read_results at C:\Users\210117\.julia\v0.4\AmplNLWriter\src\AmplNLWriter.jl:586
   in optimize! at C:\Users\210117\.julia\v0.4\AmplNLWriter\src\AmplNLWriter.jl:394
   in optimize! at C:\Users\210117\.julia\v0.4\AmplNLWriter\src\AmplNLWriter.jl:720
   in solvenlp at C:\Users\210117\.julia\v0.4\JuMP\src\nlp.jl:1246
   in solve at C:\Users\210117\.julia\v0.4\JuMP\src\solvers.jl:138
   [inlined code] from util.jl:210
   in solve at C:\Users\210117\git\GasModels2.jl\src\core/base.jl:75
   in run_generic_model at C:\Users\210117\git\GasModels2.jl\src\core/base.jl:94
   in run_gf at C:\Users\210117\git\GasModels2.jl\src\prob/gf.jl:10
   [inlined code] from C:\Users\210117\git\GasModels2.jl\test\gf.jl:7
   in anonymous at no file:0
   in include at boot.jl:261
   in include_from_node1 at loading.jl:304
   in include at boot.jl:261
   in include_from_node1 at loading.jl:304
   in process_options at client.jl:280
   in _start at client.jl:378

Is this version 4 vs. version 5 issue?

**Comment by JackDunnNZ** [comment]
This should be working, I've just tried it out on OS X and Windows without any problems on v0.4. Can we make sure the updated version is being used by running `Pkg.checkout("AmplNLWriter")` again and then restarting julia and trying again?

**Comment by rb004f** [comment]
Interesting.  I had tried that this morning with no luck, and tried again just now and now it works.  Must have had a stale compilation somewhere.  Thanks for the help.

**Comment by JackDunnNZ** [comment]
Great, glad it's working
